lrauv_ignition_plugins/src/ScienceSensorsSystem.cc

```diff
@@ -925,6 +909,15 @@ float ScienceSensorsSystemPrivate::BarycentricInterpolate(
       break;
     }
   }
+  if (ltPSortedIdx < 0 || ltPSortedIdx >= xsSortedInds.size() ||
```

Comment:
Is this happening locally for you? The sanity check above that prints
```
    ignwarn << "1D linear interpolation: query point lies on one side of all "
      "interpolation points. Cannot interpolate." << std::endl;
```
is supposed to preclude this from happening. If p is not on one side of all the points, then it has to be in between two points. If this is happening, then a case isn't being checked, and it should be checked before this for-loop, instead of checking afterwards.

Reply:
Yeah many tests are failing because of this, both locally for me and on CI. You can see how the "Failed to find consecutive elements in sorted vector." message is printed on CI multiple times. You can test with `test_controller` for example.

Reply:
Hmm I still cannot reproduce a crash. I commented out this block and the block below (diff below), and I ran almost all the `test_*` executables locally, both the standalone ones and the `test_mission_*` ones. I thought maybe you meant all the mission tests fail, but those pass locally for me too. Most of them just print the blue messages. Yoyo prints the one I saw on old CI:
```
[Wrn] [ScienceSensorsSystem.cc:878] 1D linear interpolation: query point lies on one side of all interpolation points. Cannot interpolate.
```
but it passes too.

My only diff is this:
```diff
$ git diff chapulina/barycentric_segfault
diff --git a/lrauv_ignition_plugins/src/ScienceSensorsSystem.cc b/lrauv_ignition_plugins/src/ScienceSensorsSystem.cc
index a3cfff4..f222125 100644
--- a/lrauv_ignition_plugins/src/ScienceSensorsSystem.cc
+++ b/lrauv_ignition_plugins/src/ScienceSensorsSystem.cc
@@ -196,7 +196,7 @@ class tethys::ScienceSensorsSystemPrivate
 
   /// \brief Debug printouts for interpolation. Will keep around at least until
   /// interpolation is stable.
-  public: const bool DEBUG_INTERPOLATE = false;
+  public: const bool DEBUG_INTERPOLATE = true;
 
   ///////////////////////////////
   // Variables for coordinate system
@@ -909,6 +909,7 @@ float ScienceSensorsSystemPrivate::BarycentricInterpolate(
       break;
     }
   }
+  /*
   if (ltPSortedIdx < 0 || ltPSortedIdx >= xsSortedInds.size() ||
       gtPSortedIdx < 0 || gtPSortedIdx >= xsSortedInds.size())
   {
@@ -918,6 +919,7 @@ float ScienceSensorsSystemPrivate::BarycentricInterpolate(
             << "]. Cannot interpolate." << std::endl;
     return std::numeric_limits<float>::quiet_NaN();
   }
+  */
 
   // Normalize the distances to ratios between 0 and 1, to use as weights
   float ltPWeight = ltPDist / (gtPDist + ltPDist);
@@ -927,11 +929,13 @@ float ScienceSensorsSystemPrivate::BarycentricInterpolate(
   int ltPIdx = xsSortedInds[ltPSortedIdx];
   int gtPIdx = xsSortedInds[gtPSortedIdx];
 
+  /*
   if (ltPIdx >= _values.size() || gtPIdx >= _values.size())
   {
     ignwarn << "Bad indices. Cannot interpolate." << std::endl;
     return std::numeric_limits<float>::quiet_NaN();
   }
+  */
   float result = ltPWeight * _values[ltPIdx] + gtPWeight * _values[gtPIdx];
 
   if (this->DEBUG_INTERPOLATE)
```

Reply:
Oh, `test_controller` actually does seg fault. I guess I didn't run the right one. Most of the others passed. I'll look into what case `test_controller` activates.

Reply:
It was a bug, fixed in b8641f02224ef2b9172301234277e73871919dbe
Turns out I never actually stored the sorted array, only stored the indices :sweat_smile: 
Now you should not see the bumpers printed. I changed the messages to be more precise what's happening, and added that they aren't supposed to happen so the user (future me?) will hopefully look for bugs.

Reply:
Nice! Thanks!